### PR TITLE
ST_CIC_WA_1_0 Release/feature branch

### PR DESCRIPTION
**Pull request by abul-syed**
### JIRA link (if applicable) ###

- [AM-2714](https://tools.hmcts.net/jira/browse/AM-2714) - ST_CIC_WA - Add mapping rules in ORM for ST judges
   - [AM-2717](https://tools.hmcts.net/jira/browse/AM-2717) - ST_CIC_WA - Drool JUnit testing in ORM and RAS applications for JUDICIAL users
- [AM-2716](https://tools.hmcts.net/jira/browse/AM-2716) - ST_CIC_WA - Add mapping rules in ORM for STAFF users
   - [AM-2718](https://tools.hmcts.net/jira/browse/AM-2718) - ST_CIC_WA - Drool JUnit testing in ORM and RAS applications for STAFF users
- [COT-439](https://tools.hmcts.net/jira/browse/COT-439) - Add bookable attribute to ST_CIC fee-paid-judges
- [COT-460](https://tools.hmcts.net/jira/browse/COT-460) - Update ST org roles to include region attribute

plus:
 - [AM-2866](https://tools.hmcts.net/jira/browse/AM-2866) - ST_CIC_WA - Final round of review and testing of combined feature branches for release ST_CIC_WA_1_0

### Change description ###

ST_CIC_WA_1_0 Release/feature branch:

- #1456 
- #1469 
- #1458
- #1529 
- #1546

Patch PR:

- #1514
- #1527

> NB: This PR is has a dependency on hmcts/am-role-assignment-service#1916

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolStcicStaffOrgRoleTest.java [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+                assertEquals(r.getGrantType().toString(), "BASIC");
+            }
+        });
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "1,BBA2,'senior-tribunal-caseworker,tribunal-caseworker,hmcts-legal-operations,task-supervisor,"
+                    + "case-allocator,specific-access-approver-legal-ops',Y,Y",
+            "2,BBA2,'tribunal-caseworker,hmcts-legal-operations,task-supervisor,case-allocator',Y,Y"
```
Comment (mattnayler) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolStcicStaffOrgRoleTest.java:206]: Can we have usual test with and without CaseAllocator and TaskSupervisor flags.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolStcicStaffOrgRoleTest.java [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+            }
+        });
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "9,BBA2,'ctsc-team-leader,ctsc,hmcts-ctsc,task-supervisor,case-allocator,specific-access-approver-ctsc'"
+                    + ",Y,Y",
+            "10,BBA2,'ctsc,hmcts-ctsc,task-supervisor',Y,N",
+            "10,BBA2,'ctsc,hmcts-ctsc,task-supervisor,case-allocator',Y,Y"
```
Comment (mattnayler) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolStcicStaffOrgRoleTest.java:122]: Can we have usual test with and without CaseAllocator and TaskSupervisor flags.

**Review by mattnayler — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolStcicStaffOrgRoleTest.java [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+                    + "specific-access-approver-admin',Y,N",
+            "3,BBA2,'hearing-centre-team-leader,hearing-centre-admin,hmcts-admin,case-allocator,"
+                    + "specific-access-approver-admin',N,Y",
+            "3,BBA2,'hearing-centre-team-leader,hearing-centre-admin,hmcts-admin,task-supervisor,"
+                    + "case-allocator,specific-access-approver-admin',Y,Y",
+            "4,BBA2,'hearing-centre-admin,hmcts-admin',N,N",
+            "12,BBA2,'regional-centre-team-leader,regional-centre-admin,hmcts-admin,task-supervisor,"
+                    + "case-allocator,specific-access-approver-admin',Y,Y",
+            "13,BBA2,'regional-centre-admin,hmcts-admin',N,N"
+    })
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolStcicStaffOrgRoleTest.java:37]: Resolved in 96a109309c7a0fba26d68b2193c97b29c0b25fb1

File: src/main/resources/validationrules/stcic/stcic-judicial-office-holder-mapping.drl [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+import function uk.gov.hmcts.reform.orgrolemapping.helper.AssignmentRequestBuilder.validateAuthorisation
+
+/*
+ * 1: "ST_CIC President of Tribunal-Salaried" business role mapping to JOH.
+ */
+
+rule "st_cic_president_of_tribunals_salaried_joh"
+when
+   $f:  FeatureFlag(status && flagName == FeatureFlagEnum.ST_CIC_WA_1_0.getValue())
+   $jap: JudicialAccessProfile(appointment != null,
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/resources/validationrules/stcic/stcic-judicial-office-holder-mapping.drl:34]: Resolved in 96a109309c7a0fba26d68b2193c97b29c0b25fb1

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolStcicStaffOrgRoleTest.java (hunk 1 above)
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolStcicStaffOrgRoleTest.java:206]: Resolved in #1514, see 94c4d04cf6b44c0b39ba31e417707bf90397cbe3.

File: src/main/resources/validationrules/stcic/stcic-judicial-office-holder-mapping.drl [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+import java.time.ZonedDateTime;
+import java.time.LocalDateTime;
+import uk.gov.hmcts.reform.orgrolemapping.domain.model.FeatureFlag;
+import uk.gov.hmcts.reform.orgrolemapping.domain.model.enums.FeatureFlagEnum;
+import function uk.gov.hmcts.reform.orgrolemapping.domain.service.RequestMappingService.logMsg;
+import function uk.gov.hmcts.reform.orgrolemapping.domain.service.RequestMappingService.addAndGetTicketCodes;
+import function uk.gov.hmcts.reform.orgrolemapping.helper.AssignmentRequestBuilder.validateAuthorisation
+
+/*
+ * 1: "ST_CIC President of Tribunal-Salaried" business role mapping to JOH.
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/resources/validationrules/stcic/stcic-judicial-office-holder-mapping.drl:28]: See also f285545b0d39e0e972a45a8b79ab1d84f2b20400 and 8af3f10f4ed6dfee817d9beb45ba6d4a7914b7cd.

**Review by mattnayler — APPROVED** [review]
Approved at de2ec5fc17953f8eb3633858dc714ec0c8e5025a after merger of #1514.

**Review by mikebrownccd — APPROVED** [review]
LGTM

File: src/main/resources/validationrules/stcic/stcic-judicial-org-role-mapping.drl [hunk 6]
```diff
@@ -0,0 +1,490 @@
...
+rule "st_cic_fee_paid_judge_org_role"
+when
+  $f:  FeatureFlag(status && flagName == FeatureFlagEnum.ST_CIC_WA_1_0.getValue())
+  $joh: JudicialOfficeHolder(office == "ST_CIC Tribunal Judge-Fee Paid")
+then
+   Map<String,JsonNode> attribute = new HashMap<>();
+   attribute.put("jurisdiction", JacksonUtils.convertObjectIntoJsonNode("ST_CIC"));
+   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($joh.getPrimaryLocation()));
+   attribute.put("contractType", JacksonUtils.convertObjectIntoJsonNode("Fee-Paid"));
+   attribute.put("workTypes", JacksonUtils.convertObjectIntoJsonNode("hearing_work,decision_making_work,routine_work,"
```
Comment (abul-syed) [inline src/main/resources/validationrules/stcic/stcic-judicial-org-role-mapping.drl:304]: `attribute.put("bookable", JacksonUtils.convertObjectIntoJsonNode("true"));`

`fee-paid-judge` missing bookable attribute. ExUI checks for this attribute to allow the user with role `fee-paid-judge` to create a booking.

**Review by mattnayler — COMMENTED** [review]
Re-reviewed after fc09f7bb671b4d0f50018ed4797b2c553d414011 subject to #1529.

**Review by mattnayler — APPROVED** [review]
Re-approved after merger of #1529, see d18c85a2b54d75c7e95f09661e503b04f578adc7.

**Review by mattnayler — APPROVED** [review]
Re-approved at 3e80f55f47973a1cb81b15cd385c465d5d00f028.

**Review by mattnayler — APPROVED** [review]
Re-reviewed at 9ebab30b72763ace7b52df75e94a9bb9825fe21d.

**Review by mattnayler — APPROVED** [review]
Re-approved after merger of #1546, see b98872c480179cacf1e8a89c955c2c8d0fe0bbe0.

**Review by mikebrownccd — APPROVED** [review]
LGTM

**Pull request merged**